include/zim/archive.h

```diff
@@ -254,9 +257,12 @@ namespace zim
 
       /** Check if archive has a favicon entry
        *
-       * @return True if the archive has a favicon entry.
+       * @param width The width of the illustration to check. Default to 48 (48x48px icon)
+       * @param height The height of the illustration to check. Default to 48 (48x48px icon)
+       * @return True if the archive has a corresponding illustration entry.
+       *         (Always True if the archive has no illustration, but a favicon)
        */
-      bool hasFaviconEntry() const;
+      bool hasIllustration(unsigned int width=48, unsigned int height=48) const;
```

Comment:
`archive.hasIllustration(size)` can be easily confused for `archive.hasIllustration(size, size)` whereas it actually stands for `archive.hasIllustration(size, 48)`. In such cases it is better to have two functions:

```c++
    bool hasIllustration(unsigned int width, unsigned int height) const;
    bool hasIllustration(unsigned int size=48) const { return hasIllustration(size, size); }
```